src/middlewared/middlewared/plugins/vm.py

```diff
@@ -87,7 +88,26 @@ class VMSupervisor(object):
 
     def __init__(self, manager, vm):
         self.manager = manager
-        self.logger = self.manager.logger
+
+        if not os.path.exists('/var/log/vm'):
+            os.makedirs('/var/log/vm')
+
+        self.logger = middlewared.logger.Logger(
+            f'VM: {vm["name"]}_{vm["id"]}'
+        ).getLogger()
+
+        handler = middlewared.logger.ErrorProneRotatingFileHandler(
+            f'/var/log/vm/{vm["name"]}_{vm["id"]}',
+            maxBytes=10485760,
+            backupCount=5,
+        )
+        handler.setFormatter(logging.Formatter(
+            '[%(asctime)s] (%(levelname)s) %(name)s.%(funcName)s():%(lineno)d'
```

Comment:
Maybe we should make this a constant somewhere?

Reply:
It's only used once, so I'm not sure if that's a major benefit.

Reply:
@skarekrow 
https://github.com/freenas/freenas/blob/master/src/middlewared/middlewared/logger.py#L228
https://github.com/freenas/freenas/blob/master/src/middlewared/middlewared/logger.py#L278

Reply:
Sorry I should have been more clear, in the VM plugin.

Reply:
And it should be global